### DISCUSSION TITLE
Extending nested node in tree

### DISCUSSION THREAD

**Discussion by sibuk-harabudjasim** [freeconf/restconf#50] [Q&A] [answered]
https://freeconf.org/docs/examples/node-extend/ shows example of extension of root node. 
Is there a possibility to extend node few levels deep in node tree without customizing each node along the way?
Even with custom nodes, is there an example of such extension?

More precise question:
How to define `OnChoose` for choice defined deeply in YANG tree?

**Comment by dhubler** [comment, marked as answer]
#1 nest - you can
  nodeutil.Node{}  will nest automatically and in many ways diminishes
the need to use extend.  In this example

      https://github.com/freeconf/yang/blob/master/nodeutil/schema2.go

the same code OnGetChild func is called at each level until you decide
otherwise.

OnGetChild: ...
    switch r.Meta.Ident {
         //  "target"can be 1 level or 10 levels deep
         case "target"
             // if you really want to use extend,  you can do this
             return &nodeutil.Extend{
                 Base: n.DoGetChild(r)
                  // custom code here
             }

#2 . Implement "OnChoose" for choose.  Link above has an example.

I need to update the examples to make this more obvious.

**Reply by sibuk-harabudjasim** [reply]
Thanks, I'll try with Node{}
